sound/soc/sof/sof-audio.c

```diff
@@ -412,6 +425,10 @@ sof_unprepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widg
 	struct snd_soc_dapm_path *p;
 
 	if (is_virtual_widget(sdev, widget, __func__))
+		goto sink_unprepare;
+
+	/* skip aggregated DAIs */
+	if (!swidget || is_aggregated_dai(swidget))
```

Comment:
I think this is going to cause change in behavior?

1. Previously we just returned in case of virtual_widget(), now we will proceed to sink_unprepare
2. Now we will return instead of proceeding to sink_unprepare if swidget is NULL.

Are you sure of these changes will not have side effects?